src/library/scala/util/control/ControlThrowable.scala

```diff
@@ -33,3 +33,33 @@ package util.control
  *  @author Miles Sabin
  */
 trait ControlThrowable extends Throwable with NoStackTrace
+
+object ControlThrowable {
+  sealed trait CachedBase extends Throwable with ControlThrowable {
+    override def fillInStackTrace(): Throwable = super[Throwable].fillInStackTrace()
+  }
+
+  class Cached(message: String)
```

Comment:
also, in what sense is this "cached"? It appears to just be a `Throwable` with neither stack trace nor suppression.

Reply:
It is called "cached" because it is intended to be used for cached `ControlThrowable`s, where e.g. having suppressed exceptions would lead to memory leaking. For non-cached `ControlThrowable`s (e.g. `NonLocalReturnControl`), though a suppressed exception will probably get ignored anyway, it doesn't hurt to add a suppressed exception because the `NonLocalReturnControl` will go out of scope and it will all get GCed anyway.

Reply:
I am open to other names

Reply:
Hmm, ok. I'm not a fan of the name. `Stateless` might be better, or `Immutable`, or even `Cacheable`.

My brilliant idea of "what if we just override `addSuppressed` as a no-op, too" is stymied by Java's library designers thinking they know better...

Reply:
I'm okay with `Cacheable`. `Immutable` sounds weird, cuz exceptions aren't usually thought of as mutable. Maybe `Stateless`, but I wouldn't usually think of an exception as statefull either.

I threw this together in like 15 minutes at 3am, so the name wasn't intended to be final anyway xD